docs/knn/warmup.md

```diff
@@ -0,0 +1,42 @@
+---
+layout: default
+title: Warmup API
+parent: KNN
+nav_order: 5
+has_children: false
+has_toc: false
+has_math: false
+---
+
+# Warmup API
+
+The HNSW graphs used to perform k-Approximate Nearest Neighbor Search are stored as `.hnsw` files with other Lucene segment files. In order to perform search on these graphs, they need to be loaded into native memory. If the graphs have not yet been loaded into native memory, upon search, they will first be loaded and then searched. This loading time can cause high latency during initial queries. To avoid this situation, users will often run random queries during a warmup period. After this warmup period, the graphs will be loaded into native memory and their production workloads can begin. This loading process is indirect and requires extra effort.
```

Comment:
A couple more instances of future tense that I missed before:

"If the graphs have not yet been loaded into native memory, upon search, they will first be loaded and then searched." -> "If the plugin has not yet loaded the graphs into native memory, it loads them when it receives a search request. This loading time..."

"To avoid this situation, users will often run random queries during a warmup period." -> To avoid this situation, users often run random queries...